src/pages/home/HeaderView.js

```diff
@@ -141,25 +142,37 @@ function HeaderView(props) {
         }
     }
 
-    if ((isChatThread && !isEmptyChat) || isUserCreatedPolicyRoom || canLeaveRoom) {
-        if (!isWhisperAction && props.report.notificationPreference === CONST.REPORT.NOTIFICATION_PREFERENCE.HIDDEN) {
-            threeDotMenuItems.push({
-                icon: Expensicons.ChatBubbles,
-                text: translate('common.join'),
-                onSelected: Session.checkIfActionIsAllowed(() =>
-                    Report.updateNotificationPreference(props.report.reportID, props.report.notificationPreference, CONST.REPORT.NOTIFICATION_PREFERENCE.ALWAYS, false),
-                ),
-            });
-        } else if ((isChatThread && props.report.notificationPreference.length) || isUserCreatedPolicyRoom || canLeaveRoom) {
-            const isWorkspaceMemberLeavingWorkspaceRoom = lodashGet(props.report, 'visibility', '') === CONST.REPORT.VISIBILITY.RESTRICTED && isPolicyMember;
-            threeDotMenuItems.push({
-                icon: Expensicons.ChatBubbles,
-                text: translate('common.leave'),
-                onSelected: Session.checkIfActionIsAllowed(() => Report.leaveRoom(props.report.reportID, isWorkspaceMemberLeavingWorkspaceRoom)),
-            });
-        }
+    const canJoinOrLeave = (isChatThread && !isEmptyChat) || isUserCreatedPolicyRoom || canLeaveRoom;
+    const canJoin = canJoinOrLeave && !isWhisperAction && props.report.notificationPreference === CONST.REPORT.NOTIFICATION_PREFERENCE.HIDDEN;
+    const canLeave = canJoinOrLeave && ((isChatThread && props.report.notificationPreference.length) || isUserCreatedPolicyRoom || canLeaveRoom);
+    if (canJoin) {
+        threeDotMenuItems.push({
+            icon: Expensicons.ChatBubbles,
+            text: translate('common.join'),
+            onSelected: Session.checkIfActionIsAllowed(() =>
+                Report.updateNotificationPreference(props.report.reportID, props.report.notificationPreference, CONST.REPORT.NOTIFICATION_PREFERENCE.ALWAYS, false),
+            ),
+        });
+    } else if (canLeave) {
+        const isWorkspaceMemberLeavingWorkspaceRoom = lodashGet(props.report, 'visibility', '') === CONST.REPORT.VISIBILITY.RESTRICTED && isPolicyMember;
+        threeDotMenuItems.push({
+            icon: Expensicons.ChatBubbles,
+            text: translate('common.leave'),
+            onSelected: Session.checkIfActionIsAllowed(() => Report.leaveRoom(props.report.reportID, isWorkspaceMemberLeavingWorkspaceRoom)),
+        });
     }
```

Comment:
Because we need the `canJoin` condition to show the Join button, I guess it's better to create `canLeave` too to store the leave visibility condition.